x/evm/keeper/pointer_upgrade.go

```diff
@@ -0,0 +1,108 @@
+package keeper
+
+import (
+	"math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/sei-protocol/sei-chain/utils"
+	"github.com/sei-protocol/sei-chain/x/evm/artifacts"
+	"github.com/sei-protocol/sei-chain/x/evm/state"
+	"github.com/sei-protocol/sei-chain/x/evm/types"
+)
+
+func (k *Keeper) RunWithOneOffEVMInstance(
+	ctx sdk.Context, runner func(*vm.EVM) error, logger func(string, string),
+) error {
+	stateDB := state.NewDBImpl(ctx, k, false)
+	evmModuleAddress := k.GetEVMAddressOrDefault(ctx, k.AccountKeeper().GetModuleAddress(types.ModuleName))
+	gp := core.GasPool(math.MaxUint64)
+	blockCtx, err := k.GetVMBlockContext(ctx, gp)
+	if err != nil {
+		logger("get block context", err.Error())
+		return err
+	}
+	cfg := types.DefaultChainConfig().EthereumConfig(k.ChainID(ctx))
+	txCtx := core.NewEVMTxContext(&core.Message{From: evmModuleAddress, GasPrice: utils.Big0})
+	evmInstance := vm.NewEVM(*blockCtx, txCtx, stateDB, cfg, vm.Config{})
+	err = runner(evmInstance)
+	if err != nil {
+		logger("upserting pointer", err.Error())
```

Comment:
Does this failing always mean that upserting pointer failed?

Reply:
yeah it means it failed for this particular pointer